sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/UpdatingSpecificationFilesGenerator.java

```diff
@@ -0,0 +1,253 @@
+      String checkName = rulesMappedToKey.get(rspecKey).get(0).getProvider().getMetadata().getName();
+      checkName = sanitizeCheckName(checkName, setOfExistingCheckClassNames);
+      writeCheckFile(checkName, rspecKey);
+      writeCheckTestFile(checkName);
+      newCheckNames.add(checkName);
+    }
+
+    writeUpdatedRulesDefinition(newCheckNames);
+  }
+
+  List<String> retrieveAlreadyExistingKeys() {
+    List<String> rspecKeys = new ArrayList<>();
+
+    List<Class<?>> checks = SecretsRulesDefinition.checks();
+    for (Class<?> check : checks) {
+      Constructor<?>[] declaredConstructors = check.getDeclaredConstructors();
+      Check instantiatedCheck;
+      try {
+        instantiatedCheck = (Check) declaredConstructors[0].newInstance();
+      } catch (Exception e) {
+        LOG.error("Error while retrieving already existing keys");
+        throw new RuntimeException(e);
+      }
+      rspecKeys.add(instantiatedCheck.ruleKey.rule());
+    }
+
+    return rspecKeys;
+  }
+
+  private String sanitizeCheckName(String checkName, Set<String> existingClassNames) {
+    checkName = checkName.replaceAll("[^a-zA-Z]", "");
+    checkName = checkName + "Check";
+
+    if (existingClassNames.contains(checkName)) {
+      checkName = checkName.replace("Check", "UniqueNameCheck");
+    }
+    existingClassNames.add(checkName);
+    return checkName;
+  }
+
+  private void writeCheckFile(String checkName, String rspecKey) {
+    Path checkPath = Path.of("src", "main", "java", "org", "sonar", "plugins", "secrets", "checks", checkName + ".java");
+    Path checkTemplatePath = Path.of("src", "test", "resources", "templates", "GenericCheckTemplate.java");
+    try {
+      Files.copy(checkTemplatePath, checkPath);
+
+      String content = Files.readString(checkPath, charset);
+      content = content.replaceAll("GenericCheckTemplate", checkName);
+      content = content.replaceAll("<RSPEC-KEY>", rspecKey);
+      Files.write(checkPath, content.getBytes(charset));
+    } catch (IOException e) {
+      LOG.error("Error while writing Check file with name \"" + checkName + ".java\", please fix manually", e);
+      throw new RuntimeException(e);
+    }
+
+    String successMessage = String.format("Successfully generated Check \"%s.java\" with rspecKey %s", checkName, rspecKey);
+    LOG.info(successMessage);
+  }
+
+  public void writeCheckTestFile(String checkName) {
+    Path checkTestPath = Path.of("src", "test", "java", "org", "sonar", "plugins", "secrets", "checks", checkName + "Test.java");
+    Path checkTestTemplatePath = Path.of("src", "test", "resources", "templates", "GenericCheckTestTemplate.java");
+    try {
+      Files.copy(checkTestTemplatePath, checkTestPath);
+
+      String content = Files.readString(checkTestPath, charset);
+      content = content.replaceAll("GenericCheckTemplate", checkName);
+      content = content.replaceAll("GenericCheckTemplateTest", checkName + "Test");
```

Comment:
Why `replaceAll` instead of `replace`? The first argument is not a regex, so `replace` will be faster.